gobblin-cluster/src/test/java/org/apache/gobblin/cluster/ClusterIntegrationTest.java

```diff
@@ -0,0 +1,224 @@
+import org.testng.annotations.Test;
+
+
+public class ClusterIntegrationTest {
+
+  public final static Logger _logger = LoggerFactory.getLogger(ClusterIntegrationTest.class);
+  public static final String JOB_CONF_NAME = "HelloWorldJob.conf";
+  Config _config;
+  private Path _workPath;
+  private Path _jobConfigPath;
+  private Path _jobOutputBasePath;
+  private URL _jobConfResourceUrl;
+  private TestingServer _testingZKServer;
+  private GobblinTaskRunner _worker;
+  private Thread _workerThread;
+  private GobblinClusterManager _manager;
+
+
+  @Test
+  public void simpleJobShouldComplete() throws Exception {
+    init();
+    startCluster();
+    waitForAndVerifyOutputFiles();
+    shutdownCluster();
+  }
+
+  private void init() throws Exception {
+    initWorkDir();
+    initZooKeeper();
+    initConfig();
+    initJobConfDir();
+    initJobOutputDir();
+  }
+
+  private void initWorkDir() throws IOException {
+    // Relative to the current directory
+    _workPath = Paths.get("gobblin-integration-test-work-dir");
+    _logger.info("Created a new work directory: " + _workPath.toAbsolutePath());
+
+    // Delete the working directory in case the previous test fails to delete the directory
+    // e.g. when the test was killed forcefully under a debugger.
+    deleteWorkDir();
+    Files.createDirectory(_workPath);
+  }
+
+  private void initJobConfDir() throws IOException {
+    String jobConfigDir = _config.getString(GobblinClusterConfigurationKeys.JOB_CONF_PATH_KEY);
+    _jobConfigPath = Paths.get(jobConfigDir);
+    Files.createDirectories(_jobConfigPath);
+    _jobConfResourceUrl = Resources.getResource(JOB_CONF_NAME);
+    copyJobConfFromResource();
+  }
+
+  private void initJobOutputDir() throws IOException {
+    _jobOutputBasePath = Paths.get(_workPath + "/job-output");
+    Files.createDirectory(_jobOutputBasePath);
+  }
+
+  private void copyJobConfFromResource() throws IOException {
+    try (InputStream resourceStream = _jobConfResourceUrl.openStream()) {
+      File targetFile = new File(_jobConfigPath + "/" + JOB_CONF_NAME);
+      FileUtils.copyInputStreamToFile(resourceStream, targetFile);
+    }
+  }
+
+  private void initZooKeeper() throws Exception {
+    _testingZKServer = new TestingServer(false);
+    _logger.info(
+        "Created testing ZK Server. Connection string : " + _testingZKServer.getConnectString());
+  }
+
+  private void initConfig() {
+    Config configFromResource = getConfigFromResource();
+    Config configOverride = getConfigOverride();
+    _config = configOverride.withFallback(configFromResource).resolve();
+  }
+
+  private Config getConfigOverride() {
+    Map<String, String> configMap = new HashMap<>();
+    String zkConnectionString = _testingZKServer.getConnectString();
+    configMap.put(GobblinClusterConfigurationKeys.ZK_CONNECTION_STRING_KEY, zkConnectionString);
+    configMap.put(GobblinClusterConfigurationKeys.CLUSTRER_WORK_DIR, _workPath.toString());
```

Comment:
Existing typo. It is only referenced in one other file, so can you fix it?

Reply:
Fixed. Thanks